rfcs/protocols/asset-exchange/corda-htlc-alternatives.md

```diff
@@ -0,0 +1,68 @@
+
+#### Time Window
+Corda supports the ability to constrain transactions to a defined time window. In the context of HTLCs this capability can be employed to specify the time range in which coin redemptions can occur, and the time point after which a refund claim is valid.
+
+#### Time Triggered Actions
+Corda supports the ability to schedule specific actions in a state at a specified time. In the context of HTLCs, this could be used to automatically process refunds once the HTLC expires, if claims haven't been submitted up to that point.
+
+### HTLCs in CordApp
+To create a Hashed Time Lock Contract in Corda, we need a generalised encumbrance state that encodes parameters of the HTLC agreement (i.e. time window, secret hash and identities of parties), and an associated contract which encodes the relevant validation logic (i.e. a state can be claimed by a counter party with a pre-image within a time window or redeemed by its original owner after). Each of these is briefly discussed below:
+
+#### Hash Lock Encumbrance State
+A generalised state class that encodes the properties of the HTLC such as the image of the secret, the time window within which a counter party can redeem the asset, and the public key of all involved parties as well as a reference to the binding contract logic. This state will be attached to an asset as an encumbrance, to conditionally lock the underlying asset.
+<img src="../../resources/images/asset-exchange-corda-encumbrance-lock.png" width=80%>
+
+#### Hash Lock Encumbrance Validation Contract
+The validation contract ensures that all attempts to spend the underlying asset is validated against the conditions of the hashed lock encumbrance state. That is, it ensures that only the owner or the designated counter party can make claims against the event, that the counter-party can only claim the state upon presentation of the pre-image of the stored secret within the specified validity window, and that the owner can only claim a refund of the asset if the specified time period has elapsed.
+
+#### Process
+The procedures for setting up an HTLC with one Corda network, which would need to be mirrored in the counter party network, involves the following:
+- Party A and B agree on terms of the contract (i.e. secret, time window, asset amount etc.)
+- Party A creates a transaction that takes its asset state as input and produces i) a hash lock encumbrance state and ii) the asset state with a reference to the encumbrance. Party A sends this transaction to Party B.
+- **Claiming**: Party B submits a transaction to claim the locked asset with the pre-image of the secret. If the pre-image is valid and the transaction is within the defined time window (as validated by the encumbrance contract), the output of the transaction will be a new state representing the asset with Party B as its owner that no longer has the encumbrance state attached to it.
```

Comment:
In an earlier section on this page, these bullet headings were not boldfaced. I don't care either way, but can you make them consistent on the page (either all boldfaced or none)?

Reply:
Done